nodes/src/nodes/agent_rocketride/rocketride_agent.py

```
@@ -94,67 +103,124 @@ def _run(
+
+            # The LLM signals via the remove field which memory keys it no
+            # longer needs.  We clear them from the memory store and strip them
+            # from wave history so they don't re-appear in the next prompt.
+            # This keeps the "Previous tool results" context lean as the session
+            # progresses and old intermediate results become irrelevant.
+            remove_keys = result.get('remove') or []
+            if remove_keys:
+                for key in remove_keys:
+                    try:
+                        host.memory.clear(key)
+                    except Exception as exc:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)